server/src/main/java/io/crate/types/DoubleType.java

```diff
@@ -165,7 +165,7 @@ public Double implicitCast(Object value) throws IllegalArgumentException, ClassC
             }
             return bigDecimalValue.doubleValue();
         } else if (value instanceof Number n) {
-            return n.doubleValue();
+            return Double.valueOf(String.valueOf(n)); // prevent precision errors
```

Comment:
This afaik isn't really a bug but IEEE 754 compliant widening behavior.
Or does the SQL Standard mandate this rounding behavior?

Reply:
> Or does the SQL Standard mandate this rounding behavior?

Thanks for reviewing, I could not find such information.
Initially I tried it out on Postgres but looks like I tried it wrong. I can see the same behaviour if I try with `real` instead of `float`.

I will still find out why we implicitly call float > double casts causing comparisons to fail.

Reply:
> I will still find out why we implicitly call float > double casts causing comparisons to fail.

The `1.1` from `where 1.1 = all(a)` is a `double` type causing implicit cast over `a` which is a `float_array`. The values stored are then converted from `float` > `double` with the precision discrepancies causing the comparisions to fail.

This kind of float/double mismatching can be observed from Postgres as well:
```
create table t (a real, b double precision);
insert into t values (1.1, 1.1);
select a = b from t; -- returns false
```

Btw `1.1` being a `double` is fine according to the standard:
```
<exact numeric literal> ::=
    <unsigned integer> [ <period> [ <unsigned integer> ] ]
  | <period> <unsigned integer>

22) The declared type of an <exact numeric literal> ENL is an implementation-defined exact numeric type whose scale is the number of <digit>s to the right of the <period>. There shall be an exact numeric type capable of representing the value of ENL exactly.
```